hadoop-jobs/src/main/java/com/linkedin/hadoop/jobs/HdfsWaitJob.java

```diff
@@ -0,0 +1,155 @@
+   */
+  public HdfsWaitJob(String name, Properties properties) {
+    super(new Configuration());
+    _name = name;
+    _properties = properties;
+  }
+
+  /**
+   * Method run that keeps track of the time this program has been running. If
+   * it surpasses the timeout limit, it causes the job to either fail or succeed,
+   * depending on the value of 'failOnTimeout' specified in the DSL file. This
+   * method continuously calls checkDirectory while it hasn't timed out, using
+   * sleepTime as a buffer. If checkDirectory ever returns true on any of the
+   * calls, the job succeeds.
+   *
+   * @throws Exception If there is an exception during the parameter setup
+   */
+  public void run() throws Exception {
+    String dirPath = _properties.getProperty("pathToDirectory");
+    long freshness = parseTime(_properties.getProperty("freshness"));
+    long timeout = parseTime(_properties.getProperty("timeout"));
+    long sleepTime = parseTime(_properties.getProperty("sleepInterval", "1M"));
+    long endTime = System.currentTimeMillis() + timeout;
+    boolean failOnTimeout = Boolean.valueOf(_properties.getProperty("forceJobToFail"));
+    boolean folderFound = false;
+
+    while (System.currentTimeMillis() < endTime) {
+      folderFound = checkDirectory(dirPath, freshness);
+      if (folderFound) break;
+      Thread.sleep(sleepTime);
+    }
+
+    if (!folderFound) {
+      log.info("WARNING: There were no folders in " + dirPath + " that were fresh enough.");
+      log.info("RESULT: Job timing out with parameter failOnTimeout = " + failOnTimeout);
+      if (failOnTimeout) {
+        throw new Exception();
+      }
+    }
+  }
+
+  /**
+   * Method parseTime that takes in a string, either timeout, freshness, or sleepTime.
+   * It splits the string up using whitespace as the delimiter. Then each time unit
+   * and corresponding integer is converted to its millisecond representation. The total
+   * millisecond value is then returned.
+   *
+   * @param prop The string value of either freshness, timeout or sleepTime
+   * @throws NumberFormatException
+   * @return The amount of milliseconds prop corresponds to as a long
+   */
+  public long parseTime(String prop) throws NumberFormatException {
+    long totalTime = 0;
+    String[] strArray = prop.split("\\s+");
+
+    for (int i = 0; i < strArray.length; i++) {
+      long time = Long.valueOf(strArray[i].substring(0, strArray[i].length() - 1));
+      char unit = strArray[i].charAt(strArray[i].length() - 1);
+
+      if (unit == 'S') {
+        totalTime += TimeUnit.SECONDS.toMillis(time);
+      } else if (unit == 'M') {
+        totalTime += TimeUnit.MINUTES.toMillis(time);
+      } else if (unit == 'H') {
+        totalTime += TimeUnit.HOURS.toMillis(time);
+      } else if (unit == 'D') {
+        totalTime += TimeUnit.DAYS.toMillis(time);
+      } else {
+        log.info("ERROR: Invalid input for -> " + prop + ". JOB TERMINATED.");
+        throw new NumberFormatException();
```

Comment:
Be sure to log this message as well as include it in the exception, as users will get the exception message in their Azkaban logs.